apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -72,16 +76,36 @@ public SpannerCommandTextBuilder(string commandText)
                 && !TryParseCommand(this, InsertCommand, SpannerCommandType.Insert, commandSections)
                 && !TryParseCommand(this, InsertUpdateCommand, SpannerCommandType.InsertOrUpdate, commandSections))
             {
-                if (!commandSections[0].ToUpper().StartsWith(SelectCommand))
+                if (string.Equals(commandSections[0], SelectCommand, StringComparison.OrdinalIgnoreCase))
+                {
+                    newBuilder.CommandText = commandText;
+                    newBuilder.SpannerCommandType = SpannerCommandType.Select;
+                }
+                else if (IsDdl(commandSections[0]))
+                {
+                    newBuilder.CommandText = commandText;
+                    newBuilder.SpannerCommandType = SpannerCommandType.Ddl;
+                }
+                else
                 {
                     throw new InvalidOperationException($"{commandText} is not a recognized Spanner command.");
                 }
-
-                newBuilder.CommandText = commandText;
-                newBuilder.SpannerCommandType = SpannerCommandType.Select;
             }
         }
 
+        private static bool IsDdl(string commandPart)
+        {
+            // These three form the ddl for spanner.
+            // For reference: https://cloud.google.com/spanner/docs/data-definition-language
+            return string.Equals(commandPart, CreateCommand, StringComparison.OrdinalIgnoreCase)
+                || string.Equals(commandPart, AlterCommand, StringComparison.OrdinalIgnoreCase)
+                || string.Equals(commandPart, DropCommand, StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// </summary>
+        public bool IsCreateDatabaseCommand => CommandText?.ToUpper().StartsWith(CreateDatabaseCommand) ?? false;
```

Comment:
A couple of problems here:

- Culture-sensitive upper-casing; use `string.StartsWith(string, StringComparison)`
- Whitespace: shouldn't "CREATE   DATABASE" with multiple spaces between the words work as well?

Probably fiddly to fix - I'm fine if you want to just raise an issue for now so we can come back to it.